src/dewey/indexing.clj

```diff
@@ -1,72 +1,77 @@
 (ns dewey.indexing
   "This is the logic for making changes to search index."
-  (:require [clojurewerkz.elastisch.query :as es-query]
-            [clojurewerkz.elastisch.rest.document :as es-doc]
-            [clojurewerkz.elastisch.rest :as rest]
+  (:use [slingshot.slingshot :only [try+]])
+  (:require [qbits.spandex :as s]
             [clojure-commons.file-utils :as file]
             [dewey.doc-prep :as prep]
             [dewey.config :as cfg]
-            [dewey.entity :as entity])
-  (:import [java.util Map]
-           [clojure.lang Keyword]))
+            [dewey.entity :as entity]
+            [clojure.tools.logging :as log]))
 
 
-(def ^{:private true} collection-type "folder")
-(def ^{:private true} data-object-type "file")
+(defn index-doc
+  [es doc]
+  (s/request es {:url
+                 [(cfg/es-index) :_doc (str (:id doc))]
+                 :method :put
+                 :headers {"Content-Type" "application/json"}
+                 :body doc}))
 
+(defn- update-doc
+  "Scripted updates which are only compatible with Elasticsearch 5.x and greater."
+  [es entity script params]
+  (s/request es {:url [(cfg/es-index) :_update (str (entity/id entity))]
+                 :method :post
+                 :headers {"Content-Type" "application/json"}
+                 :body {"script" {"source" script "lang" "painless" "params" params}}}))
 
-(defmulti ^{:private true} mapping-type-of type)
-
-(defmethod mapping-type-of Map
+(defn- entity-type
```

Comment:
We should probably rename this or maybe move it, given the overloading of the term "entity" -- I was confused and thought this would be for determining data-object vs. collection, not string vs. map. I can't really think of a good name, so it seems to me it might be easiest to just move this logic into the anonymous `fn` inside the `defmulti` below.